src/ReactWidget.tsx

```diff
@@ -0,0 +1,80 @@
+import { AppProps } from 'types'
+import worldId from 'index'
+import { validateInputParams } from 'utils'
+import { useEffect, useRef, useState } from 'preact/hooks'
+import styled from 'styled-components'
+import { H3, P } from 'components/text'
+
+export function ReactWidget(
+  props: AppProps & { className?: string; onSuccess?: () => void; onError?: () => void; enabled?: boolean }
+): JSX.Element {
+  const ErrorHeading = styled(H3)`
+    font-size: 16px;
+    font-weight: 500;
+    text-align: center;
+  `
+
+  const ErrorMessage = styled(P)`
+    font-size: 12px;
+    text-align: center;
+  `
+
+  const widgetReference = useRef<HTMLDivElement | null>(null)
+  const [validation, setValidation] = useState<ReturnType<typeof validateInputParams>>({ valid: false })
+
+  function activateWorldId() {
+    worldId
+      .enable()
+      .then((result) => {
+        console.log('Verified successfully:', result)
+
+        if (props.onSuccess) {
+          props.onSuccess()
+        }
+      })
+      .catch((failure) => {
+        console.warn('Verification failed:', failure)
+
+        if (props.onError) {
+          return props.onError()
+        }
+
+        activateWorldId() // Re-activate so the user can try again in case onError wasn't passed
```

Comment:
@paolodamico My idea was to prevent automatic re-enabling by passing onError function into props. Catch block will retun on line 39 after onError function call. User can pass some re-enabling logic into onError function. 

If it's too confusing I can make additional prop to preventing auto re-enabling.